### ISSUE TITLE
Warn about addressing an absent user

### ISSUE THREAD

**Issue by kurahaupo** [MCMrARM/revolution-irc#111]
One of the problems of using IRC from a phone or tablet is that typing is slow, so one is much more likely (compared with other IRC clients) to encounter the situation where you're writing something to another user in a channel, and they leave or quit before you hit send.

Ideally I'd like to have, as soon as you've fully mentioned any nick at the start of a message (with nick+colon) to check whether the user is currently in the channel, and to keep checking as the message is being typed.

If at any time while you're typing, they leave or quit, give a strong visual warning; I suggest turning the text entry area red, if that's not too hard to implement.

Then I could immediate stop wasting effort typing a message that the intended recipient will never see.

If the same behaviour could be done in PM sessions, that would be a bonus, but I suspect it would be much more difficult without "leave" or "quit" notifications from the server.

**Comment by MCMrARM** [comment]
Nope, not going to do any messing with the background color. However, a 'this person is offline' of some sort could be added.

**Comment by Mikaela** [comment]
WeeChat has an option of changing colour of nicks that have left the channel.

(`weechat.look.color_nick_offline` & `weechat.color.chat_nick_offline`)

**Comment by MCMrARM** [comment]
Changing nick colour is an option, but I'm not sure if I like the idea - I don't think messages that already were received should be manipulated in any way.

**Comment by kurahaupo** [comment]
Changing the colour of nicks or other parts of already-displayed messages wouldn't help this problem.

When using a touch screen to type, one's visual focus is constrained to the "keyboard" and text entry area. A signal that requires looking anywhere else would be completely useless.

It doesn't have to be the text entry area that changes colour, but it has to be SOMETHING LARGE AND OBVIOUS in one's peripheral vision, if it's to be any use at all.

**Comment by kurahaupo** [comment]
Is changing the background colour of the input text area technically difficult or stylistically objectionable?

How about changing the background colour of the title of a pm tab when its relevant nick is online or offline or undecidable?

How about changing the colour of the Send button when in a pm or when the composed text starts with *nick:*.

**Comment by ZeroKnight** [comment]
> How about changing the colour of the Send button when in a pm or when the composed text starts with _nick:_.

This actually sounds like it would work pretty well; the send button is right within one's visual area when looking at the keyboard. You could change it to some kind of accent color. I think also adding a "user has gone offline" pseudo-message would be a good idea as well, as new users would be likely to understand the reason behind the button color changing.
